File: 27/main.ts
function removeElement(nums: number[], val: number): number {
  let count = 0;
  let slow_pointer = 0;
  for (let i = 0; i < nums.length; i++) {
    nums[slow_pointer] = nums[i];
    if (nums[i] === val) {
      count++;
    } else {
      slow_pointer++;
    }
  }
  return nums.length - count;
}

let a = [3, 2, 2, 3];
console.log(removeElement(a, 3));
console.log(a);

File: 27/main.py
from typing import List


class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        """
        * 2 pointers
        * one for where to put, one for where to look
        * end when look pointer reaches end
        * k = slow
        """
        slow, fast = 0, 0
        while fast < len(nums):
            nums[slow] = nums[fast]
            if nums[fast] != val:
                slow += 1
            fast += 1
        return slow
